### PR TITLE
Migrate

### PR DESCRIPTION
**Pull request by guidotijskens**
updated oci-image-migrate code
change the configuration data, moved it into a config.yaml file
code is tested on OL7/python 2.7
the rpm builds on OL8/python 3.6, tests are limited as nbd kernel module is missing on OL8; looking for a workaround/alternative
tested upload of various qcow2 and vmdk images to the OCI object storage 
.

### CODE REVIEW

File: lib/oci_migrate/migrate/gen_tools.py [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+import six
+from oci_migrate.migrate.exception import NoSuchCommand
+from oci_migrate.migrate.exception import OciMigrateException
+
+logger = logging.getLogger('oci-utils.oci-image-migrate')
+debugflag = False
+verboseflag = False
+thistime = datetime.now().strftime('%Y%m%d%H%M')
+resultfilename = '/tmp/oci_image_migrate_' + thistime
+nameserver = '8.8.8.8'
```
Comment (wbora) [inline lib/oci_migrate/migrate/gen_tools.py]: Why are we hard coding the DNS server to Google?

File: data/oci-migrate-conf.yaml [hunk 2]
```diff
@@ -0,0 +1,272 @@
+# ----------------------------------------------------------------------------
+# Configuration and status data with respect to the oci-image-migrate.
+# ----------------------------------------------------------------------------
+#
+# dummy format key
+dummy_format_key : '01234567'
+#
+# python version
+pythonver : 2
```
Comment (wbora) [inline data/oci-migrate-conf.yaml]: I am not following what we doing here, on OL7 we use python 2 on OL8 we use python 3 - python.six should allow you to not care which one you using...

File: lib/oci_migrate/migrate/doc/oci_image_migrate.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
+# oci-utils.oci-image-migrate
```
Comment (wbora) [inline lib/oci_migrate/migrate/doc/oci_image_migrate.md]: This seems more of a developer type doc than a usage/overview one, maybe split this up moving the developer specific content into a developer file

File: data/oci-migrate-conf.yaml (hunk 2 above)
Comment (wbora) [inline data/oci-migrate-conf.yaml]: I am not following what we doing here, on OL7 we use python 2 on OL8 we use python 3 - python.six should allow you to not care which one you using...

File: lib/oci_migrate/migrate/gen_tools.py [hunk 4]
```diff
@@ -0,0 +1,671 @@
...
+        bool: True on success, False otherwise.
+    """
+    logger.debug('Testing thread.')
+    if threadid in threading.enumerate():
+        return True
+    else:
+        return False
+
+
+class ProGressBar(threading.Thread):
```
Comment (wbora) [inline lib/oci_migrate/migrate/gen_tools.py]: Class name looks a bit strange, Guess you meant ProgressBar...

File: lib/oci_migrate/migrate/gen_tools.py [hunk 5]
```diff
@@ -0,0 +1,671 @@
...
+        Length of the progress bar.
+    iinterval: float
+        Interval in sec of change.
+    progarr: list
+        List of char or str to use; the list is mirrored before use.
+    """
+    def __init__(self, llength, iinterval, progarr=None):
+        self._stopthread = threading.Event()
+        threading.Thread.__init__(self)
+        self.llen = llength
```
Comment (wbora) [inline lib/oci_migrate/migrate/gen_tools.py]: Privates should be _<name>

File: lib/oci_migrate/migrate/gen_tools.py [hunk 6]
```diff
@@ -0,0 +1,671 @@
...
+        self.iint = iinterval
+        if progarr is None:
+            self.prog2 = ['-', '/', '|', '\\']
+        else:
+            self.prog2 = progarr
+            lll = len(progarr)
```
Comment (wbora) [inline lib/oci_migrate/migrate/gen_tools.py]: I normally do not care too much about var names, but these are just too much out there, we still need to make this maintainable by other people...

File: lib/oci_migrate/migrate/gen_tools.py [hunk 7]
```diff
@@ -0,0 +1,671 @@
...
+        sys.stdout.write('  [%s]\r  [' % empty)
+        sys.stdout.flush()
+        i = 0
+        j = i%ll
+        k = 0
+        while True:
+            sys.stdout.write('%s' % self.prog2[j])
+            sys.stdout.flush()
+            k += 1
+            if k == self.llen:
```
Comment (wbora) [inline lib/oci_migrate/migrate/gen_tools.py]: Same comment as above, the variable names should really be a lot more clear...

File: lib/oci_migrate/migrate/imgdevice.py [hunk 8]
```diff
@@ -0,0 +1,1402 @@
...
+        """
+        Initialisation of the UpdateImage object.
+
+        Parameters:
+        ----------
+            vdiskdata: dict
+                Information about the virtual image as described in the
+                readme.txt, img_info.
+        """
+        self.imgdata = vdiskdata
```
Comment (wbora) [inline lib/oci_migrate/migrate/imgdevice.py]: Private attribute should be called self._<name>

File: data/oci-migrate-conf.yaml [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+    lsblk      : util-linux
+    sfdisk     : util-linux
+    fdisk      : util-linux
+    parted     : parted,
+    partx      : util-linux
+    lvmdiskscan: lvm2
+    oci        : oci-cli
+#
+# logging configuration
+logging_config:
```
Comment (ejannett) [inline data/oci-migrate-conf.yaml]: Why are we defining again logging ? is it not based on logging we already have in oci_utils ?

File: lib/oci_migrate/migrate/__init__.py [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# #!/usr/bin/python
```
Comment (ejannett) [inline lib/oci_migrate/migrate/__init__.py]: We do not need this . for support of python3 we should remove such statement

**Comment by guidotijskens** [comment]
outdated

**Pull request closed without merging**